src/http/heila-http.js

```diff
@@ -0,0 +1,91 @@
+import * as heilaCore from '../core/heila-core';
+import {throwStatus, createJsonRoute} from '../util/express';
+import {assert} from '../validation';
+
+const deleteHeila = createJsonRoute(function(req, res) {
+  const uuid = req.params.uuid;
+  return heilaCore.deleteHeila(uuid)
+    .then(rowsInserted => undefined)
+    .catch(err => {
+      console.log('error happened while deleteHeila was called');
+      console.log(err);
+      throwStatus(500, 'Something went wrong.');
+    })
+});
+
+const putHeila = createJsonRoute(function(req, res) {
+  const heila = assert(req.body, 'heila');
+  return heilaCore.createOrUpdateHeila(heila)
+    .then(rowsInserted => {
+      if (rowsInserted === -1) {
+        console.log('something went wrong in putHeila');
+        throwStatus(500, 'Something went wrong.');
+      }
+    });
+});
+
+const getHeilaList = createJsonRoute(function(req, res) {
+  const userId = req.query.userId;
+  const uuid = req.params.uuid;
+
+  // if there's a query param for userId, then return only that
+  if (userId) {
+    //console.log('get single heila details by userId' + userId);
+    return heilaCore.getHeilaByUserId(userId)
+    .then(heila => {
+      if (heila === null) {
+        const err = new Error('Heila not found: ' + userId);
+        err.status = 404;
+        throw err;
```

Comment:
Use `throwStatus` here. We don't want the client to have access to stack trace etc.